docs/build.md

```diff
@@ -227,6 +229,47 @@ spec:
       name: icr-knbuild
 ```
 
+### Runtime-Image
+
+Runtime-image is a new image composed with build-strategy outcomes. On which you can compose a multi-stage build action, copying parts out the original image into a new one. This feature allows replacing the base-image of any container-image, creating final leaner images, and other use-cases that require multi-stage build.
+
+The following examples illustrates how to use this feature:
+
+```yml
+apiVersion: build.dev/v1alpha1
+kind: Build
+metadata:
+  name: nodejs-ex-runtime
+spec:
+  strategy:
+    name: buildpacks-v3
+    kind: ClusterBuildStrategy
+  source:
+    url: https://github.com/sclorg/nodejs-ex.git
+  output:
+    image: image-registry.openshift-image-registry.svc:5000/build-examples/nodejs-ex
+  runtime:
+    base:
+      image: docker.io/node:latest
+    workDir: /home/node/app
+    directories:
+      - /workspace/source:/home/node/app
```

Comment:
I wonder how a user knows which source directories to take. I mean the user does not know how the build strategy looks like = how would he know where his sources got extracted and built.

Reply:
Depends entirely in the build-strategy builder. So `s2i` may create a different image structure than `paketo` and therefore, the user needs investigate the images and decide what needs to be copied. This feature is meant for an advanced use-case.

Reply:
Hm, not 100 % buying this. I remember that in a community call I brought up that this looks like multi-stage build and it might have been @gabemontero or @sbose78 who mentioned that in build v1, the runtime image support was added before Docker had multi-stage support and that multistage is for advanced Dockerfile users. So, this feels like we add another advanced users use case, which might be fine ...

Reply:
If a user already has a multi-stage Dockerfile which she is happy with, the user should not be using runtime base images.

Also, the user may have a multi-stage Dockerfile Or a buildpacks build with runtime base images in them which takes care of lean images. 

However, the final base image may not be the one that adheres to company policies, as an example. 

In such a scenario, this feature ensures that ***no matter*** what the base image specified by the user's single/multi-stage Dockerfile is, this final image is always going to have a specific enterprise-mandated base image.

@qu1queee ^ You had a concern about how this would matter in case of Buildpacks builds, I've tried to address it here. Could you please check?

Reply:
@sbose78 I understand it for buildpacks. For a Dockerfile build, if she already understands multi-stage, she could simply take the company policy one already (assuming she has control over the source code).

Regarding my original concern about the directory structure. Is there a reasonable way to provide some variable like `$workspace` or `$workDir` through some means? With a value of `path.join('/workspace/source', build.spec.source.contextDir)` ?

Reply:
@sbose78 yes, that was clear, danke.

Reply:
> Regarding my original concern about the directory structure. Is there a reasonable way to provide some variable like `$workspace` or `$workDir` through some means? With a value of `path.join('/workspace/source', build.spec.source.contextDir)` ?

I think that's a good idea @SaschaSchwarze0, would make things simpler for the final user.

Reply:
@SaschaSchwarze0 the variable with the working directory path (`/workspace/source`) is already present on [`$(build.source.contextDir)` variable](https://github.com/redhat-developer/build/blob/17fa2792cc14288adea0bafa624e85b6db7232e0/pkg/controller/buildrun/generate_taskrun.go#L37).

Reply:
@otaviof I am not understanding your comment. Using `build.spec.source.contextDir`, the build creator can optionally specify a directory within the source repository that is to be built. What envision for your runtime support is something like this:

```Dockerfile
  runtime:
    paths:
      - ${workspace}/build/target.jar:/app/target.jar
```

Here, the value of ${workspace} will need to be substituted in the generated `Dockerfile.runtime` by the evaluation result of `path.Join('/workspace/source', build.spec.source.contextDir)`. This helps the user because he knows which value in `build.spec.source.contextDir` he has set, but he does not know that we use `/workspace/source` to clone the repo (this is defined partially by Tekton, which afaik has `/workspace` hard-coded and the `source` path comes from https://github.com/redhat-developer/build/blob/37ff3520f223fd26cb9a39d8828acf1d07a1115d/pkg/controller/buildrun/generate_taskrun.go#L52-L57 and https://github.com/redhat-developer/build/blob/37ff3520f223fd26cb9a39d8828acf1d07a1115d/pkg/controller/buildrun/generate_taskrun.go#L19).

Reply:
@SaschaSchwarze0 thanks for the explanation, indeed the `$(workspace)` takes a different shape when using the `ContextDir`. I've changed the implementation in the last commit, adding a `$(workspace)` placeholder, please consider.

Reply:
Cool @otaviof. Code looks good. Can you please update the documentation? The sample should then be `$(workspace):/home/node/app`  instead of `/workspace/source:/home/node/app`. And the documentation of `paths` should contain a sentence like *You can use the `$(workspace)` token to access the directory where your source repository is cloned to. If you provided a `spec.source.contextDir`, then `$(workspace)` points to that directory within the cloned repository.*

Reply:
Please consider the latest commit.